nym-api/src/unstable_routes/account.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use crate::{
+    node_status_api::models::{AxumErrorResponse, AxumResult},
+    support::http::state::AppState,
+    unstable_routes::models::{
+        NymVestingAccount, NyxAccountDelegationDetails, NyxAccountDelegationRewardDetails,
+        NyxAccountDetails,
+    },
+};
+use axum::{
+    extract::{Path, State},
+    routing::get,
+    Json, Router,
+};
+use cosmwasm_std::{Coin, Decimal};
+use nym_topology::NodeId;
+use nym_validator_client::nyxd::AccountId;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, str::FromStr};
+use tracing::{error, instrument, warn};
+use utoipa::ToSchema;
+
+pub(crate) fn routes() -> Router<AppState> {
+    Router::new().route("/:address", get(address))
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, ToSchema, utoipa::IntoParams)]
+pub struct AddressQueryParam {
+    #[serde(default)]
+    pub address: String,
+}
+
+#[utoipa::path(
+    tag = "Unstable",
+    get,
+    path = "/{address}",
+    context_path = "/v1/unstable/account",
+    responses(
+        (status = 200, body = NyxAccountDetails)
+    ),
+    params(AddressQueryParam)
+)]
+#[instrument(level = "info", skip_all, fields(address=address))]
+async fn address(
+    Path(AddressQueryParam { address }): Path<AddressQueryParam>,
+    State(state): State<AppState>,
+) -> AxumResult<Json<NyxAccountDetails>> {
+    let account_id = AccountId::from_str(&address).map_err(|err| {
+        error!("{err}");
+        AxumErrorResponse::not_found(&address)
+    })?;
+
+    let node_cache = state.nym_contract_cache();
+    let nym_nodes = node_cache
+        .nym_nodes()
+        .await
+        .into_iter()
+        .map(|node| (node.node_id(), node))
+        .collect::<HashMap<_, _>>();
+
+    let state_client = state.nyxd_client;
+    let mut total_value = 0u128;
+
+    // 1. get balances of chain tokens
+    let balances = state_client
```

Comment:
if we're only concerned about `unym`, query for only `unym` balance rather than getting `all_balances`. this way you'd also avoid having to mess with `Vec` and adding up amounts to `total_value`